File: head_to_offer/46/hto_46.py
class Solution:
    def translateNum(self, num: int) -> int:
        a = b = 1
        y = num % 10
        while num != 0:
            num //= 10
            x = num % 10
            a, b = (a + b if 10 <= 10 * x + y <= 25 else a), a
            y = x
        return a

File: hto/46/hto_46.cpp
class Solution {
public:
    int translateNum(int num) {
        string s = to_string(num);
        long a = 1, b = 1;
        for (int i = 2; i <= s.size(); i++) {
            string tmp = s.substr(i - 2, 2);
            long c = (tmp >= "10" && tmp <= "25") ? a + b : a;
            b = a;
            a = c;
        }
        return a;
    }
};